### PR TITLE
payout api implementation

### PR DESCRIPTION
**Pull request by michaelngangom**
**Description**
payout api implementation

**Tested scenarios**
Tested Payout Endpoints based on Mocks.

### CODE REVIEW

File: src/test/java/com/adyen/PayoutTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        assertEquals("8515136787207087", result.getPspReference());
+        assertEquals("8415088571022720", result.getRecurringDetailReference());
+    }
+
+    @Test
+    public void testConfirmThirdPartySuccess() throws Exception {
+        Client client = createMockClientFromFile("mocks/payout/modifyResponse-success.json");
+        Payout payout = new Payout(client);
+
+        ModifyRequest request = new ModifyRequest();
```
Comment (Aleffio) [inline src/test/java/com/adyen/PayoutTest.java]: Can you please create a ConfirmThirdPartyRequest that just extends ModifyRequest?
It is easier for the merchants to just refer to a ConfirmThirdPartyRequest when calling payout.confirmThirdParty.

File: src/test/java/com/adyen/PayoutTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        ModifyResponse result = payout.confirmThirdParty(request);
```
Comment (Aleffio) [inline src/test/java/com/adyen/PayoutTest.java]: Can you please do the same as ConfirmThirdPartyRequest with ConfirmThirdPartyResponse?

File: src/test/java/com/adyen/PayoutTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        assertEquals("GREEN", result.getAdditionalData().get(FRAUD_RESULT_TYPE));
+        assertEquals("false", result.getAdditionalData().get(FRAUD_MANUAL_REVIEW));
+    }
+
+    @Test
+    public void testDeclineThirdPartySuccess() throws Exception {
+        Client client = createMockClientFromFile("mocks/payout/modifyResponse-success.json");
+        Payout payout = new Payout(client);
+
+        ModifyRequest request = new ModifyRequest();
```
Comment (Aleffio) [inline src/test/java/com/adyen/PayoutTest.java]: Can you please do the same as ConfirmThirdPartyRequest with DeclineThirdPartyRequest?

File: src/test/java/com/adyen/PayoutTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        ModifyResponse result = payout.declineThirdParty(request);
```
Comment (Aleffio) [inline src/test/java/com/adyen/PayoutTest.java]: Can you please do the same as ConfirmThirdPartyRequest with DeclineThirdPartyResponse?

File: src/main/java/com/adyen/model/payout/SubmitRequest.java [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+    @SerializedName("shopperReference")
+    private String shopperReference = null;
+
+    @SerializedName("shopperStatement")
+    private String shopperStatement = null;
+
+    @SerializedName("socialSecurityNumber")
+    private String socialSecurityNumber = null;
+
```
Comment (Aleffio) [inline src/main/java/com/adyen/model/payout/SubmitRequest.java]: Can you please add the constructor with: 

Recurring recurring = new Recurring();
recurring.setContract(Recurring.ContractEnum.PAYOUT);
this.recurring = recurring;

Like you did in the other files?

File: src/main/java/com/adyen/model/payout/ModifyRequest.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    public void setAdditionalData(Map<String, String> additionalData) {
+        this.additionalData = additionalData;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("class ModifyRequest {\n");
+        sb.append("class ModifyRequest {\n");
```
Comment (lancergr) [inline src/main/java/com/adyen/model/payout/ModifyRequest.java]: duplicate

File: src/main/java/com/adyen/model/payout/ModifyResponse.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    public void setResponse(String response) {
+        this.response = response;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("class ModifyRequest {\n");
```
Comment (lancergr) [inline src/main/java/com/adyen/model/payout/ModifyResponse.java]: ModifyResponse

**Pull request merged**